app/templates/components/wds-global-navigation.hbs

```diff
@@ -54,6 +54,7 @@
 	</div>
 	{{#if model.partner_slot}}
 		<div class="wds-global-navigation__partner-slot">
+			<span class="wds-global-navigation__partner-slot-horizontal-bar"></span>
```

Comment:
I'd much rather wrap partner slot into another div to make the border work than add an empty span, because the rule of thumb is to avoid empty elements in html.

If we absolutely need to use a separate element to make border top happen, let's use a div instead of span - it has display: block by default, so you won't need to add it in css.